GPT/gpt-cursorless.talon

```diff
@@ -9,4 +9,7 @@ tag: user.cursorless
     result = user.gpt_apply_prompt(user.staticPrompt, text)
     user.cursorless_insert(cursorless_destination, result)
 
-
+# Apply a prompt to any text, and output it any target
+^model go {user.staticPrompt} <user.cursorless_target> <user.cursorless_destination>$:
```

Comment:
do we want the same set of prompts here or do we want prompts specific to `go`?

Reply:
I don't think there's a need to keep the prompts separate. This goes back to the point that we are not intending to do function calling here; we're simply allowing the user to ask anything and let the LLM make a call as to what to do in response.

Reply:
Seems fine for now. I also wonder if we want target and/or destination to be optional

Reply:
But I guess all of these prompts expect input text? So I guess makes sense to require both

Reply:
I could see an argument to make destination optional. In that case it would just paste the response instead of putting it into the destination in case of an insertion function call.

Reply:
Could you give me an example of the Python code to handle the optional destination there? I'm not sure exactly how it would appear to Python if the user doesn't supply it.

Reply:
Thinking on this a bit more, I imagine that some prompts will only makes sense with function calling. Eventually we may want to split out a separate set of prompts for those. I'd like to support both the existing ones and some additional ones if we get there, but this is probably a future concern.

Reply:
> Thinking on this a bit more, I imagine that some prompts will only makes sense with function calling. Eventually we may want to split out a separate set of prompts for those. I'd like to support both the existing ones and some additional ones if we get there, but this is probably a future concern.

yeah i think we can prob address that later if we come up with examples



> Could you give me an example of the Python code to handle the optional destination there? I'm not sure exactly how it would appear to Python if the user doesn't supply it.

You could either split it into two different commands, or wrap the destination in `[]` and then pass `cursorless_destionation or 0` as the arg to the action

Reply:
Thanks for that-I was able to make it optional. I think that makes it a lot more powerful.